test/upgrade/upgrade.go

```diff
@@ -113,3 +116,31 @@ func createTestNamespace(t *testing.T, ns string) {
 		t.Fatalf("Couldn't create namespace %q: %v", ns, err)
 	}
 }
+
+func streamFromTestNamespace(t *testing.T, clients *test.Clients, serviceName string) logstream.Canceler {
```

Comment:
Alternatively we could have a `streamFromNamespace` func passing some stream options but I guess this helps hide some details.

Reply:
Yeah. The options would have to be passed on several places so the code would be longer overall. And the options have this type: `opts ...func(*logSource)`  The logSource can't be used outside of the logstream package.

Reply:
Alternatively, I could merge the two functions because we're not going to turn on/off one or the other, we'll always turn them on together. It could look like this:
```
func streamLogs(t *testing.T, clients *test.Clients, serviceName string) logstream.Canceler {
	testStream := logstream.New(context.Background(), clients.KubeClient,
		logstream.WithNamespaces(test.ServingFlags.TestNamespace),
		logstream.WithLineFiltering(false),
		logstream.WithPodPrefixes(serviceName))
	testStreamCancel, err := testStream.StartStream(serviceName, t.Logf)
	if err != nil {
		t.Fatalf("Unable to stream logs from test namespace %s: %v", test.ServingFlags.TestNamespace, err)
	}
	
	sysStream := logstream.New(context.Background(), clients.KubeClient,
		logstream.WithNamespaces(strings.Split(system.Namespace(), ",")...))
	sysStreamCancel, err := sysStream.StartStream(serviceName, t.Logf)
	if err != nil {
		t.Fatal("Unable to stream logs from system namespaces", err)
	}

	return func() {
		testStreamCancel()
		sysStreamCancel()
	}
}
```
And the calling function would just call this:
```
if !test.ServingFlags.DisableLogStream {
	canceler = streamLogs(c.T, clients, names.Service)
}
```